### ISSUE TITLE
Featherlight lvl 281 fails to extract in 0.26.3+

### ISSUE THREAD

**Issue by class101** [cadon/ARKStatsExtractor#653]
Hello

I have checked everything of your tips but nothing could be done to import this Featherlight under the lastest SB versions. Rev erting to 0.25.3 solves the problem
![20180326192804_1](https://user-images.githubusercontent.com/1810484/37922160-2486982a-312c-11e8-9f5c-857704f9917b.jpg)

---
Steps to reproduce :

- go in Experimental OCR > Output > Enable logging and Editing
- Drag my picture to the box
- Go in Extract
- Hit Bred
- Hit extract 

notice the bug

Food is red and speed level 0

- In food select 2639,3
- Delete
- Type manually 2639,3
- Extract again

It works this time !

---

Official PvE Aberration 218
Born level 281 (best expected 283)
Breeding
Female
Just hatched, no imprint
If needed to check real time, you can reach me on the server I have 50+ Featherlight high level there, the one in the screenshot is my top 2 because I expect a level 283 soon.
values.json up to date, still works in 0.25.3 with the new json file so I don't think it has something to do with it
Checked more closely the versions, it seems the last version able to extract it is 0.26.0.

---

0.25.3

![image](https://user-images.githubusercontent.com/1810484/37922305-7e45507c-312c-11e8-9b77-2cdd1230ae19.png)

0.26.3 / 0.27.0

![image](https://user-images.githubusercontent.com/1810484/37922392-b530d976-312c-11e8-9c65-2caf070f335d.png)

---

Default values except the screen capture settings adapted for 2660x1440 144hz
![image](https://user-images.githubusercontent.com/1810484/37923630-e0f9d83e-312f-11e8-92ce-b1e2a0fb28cf.png)
![image](https://user-images.githubusercontent.com/1810484/37924772-27406ad0-3133-11e8-8afc-4ae776b8cb13.png)

**Comment by class101** [comment]
Level 281
Breeding
Female
Just hatched, no imprint

**Comment by class101** [comment]
0.25.3

![image](https://user-images.githubusercontent.com/1810484/37922305-7e45507c-312c-11e8-9b77-2cdd1230ae19.png)

0.26.3 / 0.27.0

![image](https://user-images.githubusercontent.com/1810484/37922392-b530d976-312c-11e8-9c65-2caf070f335d.png)

**Comment by class101** [comment]
![image](https://user-images.githubusercontent.com/1810484/37923630-e0f9d83e-312f-11e8-92ce-b1e2a0fb28cf.png)

**Comment by cadon** [comment]
Hi, I can't reproduce the issue, it works for me in 0.27.0.

**Comment by class101** [comment]
Well there I extracted 0.27.0 in a new folder, changed to 2560x1440, still same error, speed lvl to 0 and food is red.

Not sure how you have your test working but this is serious problem you have if you can't reproduce bug reports

**Comment by class101** [comment]
ok found the steps to reproduce:

- go in Experimental OCR > Output > Enable logging and Editing
- Drag the box
- Go in Extract, notice the bug

Food is red and speed level 0

- In food select 2639,3
- Delete
- Type manually 2639,3
- Extract again

It works this time !

I think it lacks test on this extraction method if you did not see it, that dangerous to leave it untested

**Comment by cadon** [comment]
Hi, thanks for the elaborate report! I didn't mean to be rude with that previously short answer, but it just worked for me without issues. I didn't use the OCR, but just typed in the values, not thinking that could be a difference. But indeed that was the issue. So thanks again for the steps to reproduce the issue.
The problem was that the OCR used float-variables for the read stat-values. They are usually not very large and only have one decimal, so float worked well. On close inspection, the food-value as a float was `2639.3` (as expected), but as soon as it was casted to a double and stored in the stat-input (where you also can enter the stat-values), it changed to `2639.3000488`. This aberration caused the extraction to fail. The change I made in 0.27 was optimizing the extraction-algorithm, so the thresholds are very narrow (to filter out as much as possible of the wrong results).
This issue will be fixed in the next release, I also added this featherlight to the test-cases the tool runs before each release (I don't have test-units for the OCR yet, didn't think I'd need them), the food stat seems to be just on the edge of creating rounding-issues, so it's a good indicator if the algorithm is still working.

On another note, did you try the creature-export function already? Maybe that's faster than using the OCR (it will always yield the correct values).
